engine/common/requester/engine.go

```diff
@@ -80,14 +81,20 @@ func New(log zerolog.Logger, metrics module.EngineMetrics, net module.Network, m
 		return nil, fmt.Errorf("invalid retry maximum (must not be smaller than initial interval)")
 	}
 
-	// make sure we don't send requests from self or unstaked nodes
+	// make sure we don't send requests from self
 	selector = filter.And(
 		selector,
-		filter.HasStake(true),
-		filter.Not(filter.Ejected),
 		filter.Not(filter.HasNodeID(me.NodeID())),
 	)
 
+	// make sure we don't send requests to unstaked nodes
+	if cfg.ValidateStaking {
+		selector = filter.And(
+			selector,
+			filter.HasStake(true),
+			filter.Not(filter.Ejected))
```

Comment:
I like the idea of ejection being a mechanism that works irrespective of whether you're staked or not.

Reply:
I'm not sure what you mean...

Reply:
I would apply `filter.Not(filter.Ejected)` outside of the conditional `if cfg.ValidateStaking`

Reply:
Ah right, moved so its applies always